fix(analyzer): ignore type-only imports & exports

buschtoens opened the pull request:
Prevents [type-only imports / exports](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-8.html#type-only-imports-and-export) from ending up in the final bundle.

```ts
import type { compute } from 'heavy-library';

export async function performHeavyComputation(
  ...args: Parameters<typeof compute>
): Promise<ReturnType<typeof compute>> {
  const { compute } = await import('heavy-library');
  return compute(...args);
}
```

ef4 commented:
Thanks for working on this, this will be nice.

boris-petrov commented:
Does this fix [this issue](https://github.com/ef4/ember-auto-import/issues/169#issuecomment-603088345) (i.e. this comment in that issue)?

ef4 commented:
Yes, I think this would fix that.

jamescdavis commented:
Bump. 🙂

buschtoens commented:
Alright, sorry that I kept this idling for so long, but I had no clue and time to dig into, why the 4 `*-own` scenarios were failing, while all other were running successfully. I finally got around to it now!

The issue was that _all_ scenarios, except for `*-own`, seem to directly or indirectly enable the TypeScript syntax in the babel options. Without the TS syntax enabled, TS constructs like `import type ...` cannot be parsed, which is obvious in hindsight, but was quite annoying to debug, as there was no apparent syntax error getting thrown. 😅

I've added `@babel/plugin-syntax-typescript` to the `babelOptions` in the `beforeEach` hook via 144005204bef33779fc7720dc2fe63b6e2971fa3, which fixes the issue. I don't really know, whether this solution is acceptable / ideal though, or whether I should opt-in to TS somewhere else.
Any pointers would be greatly appreciated!

I've also rebased the PR and it's all working fine now. ✅

Please let me know, whether I should add more tests, or whether I should change the TS syntax opt-in.

ef4 commented:
Looks great, thanks.
